test/search_index_ddoc_update_tests.erl

```diff
@@ -0,0 +1,277 @@
+          upload_ddoc(Host, DbName, UpdatedDDocJsonObj),
+          ensureIndexIsClosed(DbName, "_design/searchddoc", "other", Sig)
+    end)}.
+
+should_stop_indexing_after_deleting_ddoc(PortType, {Host, DbName}) ->
+    {timeout, 60,
+     ?_test(begin
+         ok = meck:reset(clouseau_rpc),
+         ok = meck:reset(dreyfus_index),
+         ok = create_doc(PortType, ?l2b(DbName), <<"doc_id">>, {[]}),
+         ReqUrl = Host ++ "/" ++ DbName ++ "/_design/searchddoc/_search/other?q=*:*",
+         {ok, Status, _Headers, _Body} =
+              test_request:get(ReqUrl, [?AUTH]),
+          ?assertEqual(200, Status),
+          DDocId = "_design/searchddoc",
+          Sig = ensureIndexIsOpened(DbName, DDocId, "other"),
+
+          % Delete the design doc and verify that the index will be closed
+          delete_ddoc(Host, DbName, ?l2b(DDocId)),
+          ensureIndexIsClosed(DbName, "_design/searchddoc", "other", Sig)
+    end)}.
+
+updated_ddoc_json_obj(DbName, DDocId) ->
+    {ok, DDoc} = fabric:open_doc(DbName, <<DDocId/binary>>, [ejson_body, ?ADMIN_CTX]),
+    % {DDocJsonObj} = ?DDOC_UPDATE,
+    % Rev = {<<"_rev">>, couch_doc:rev_to_str(DDoc#doc.revs)},
+    % DDocUpdated = {[Rev] ++ DDocJsonObj},
+    % DDocUpdated.
+    UpdatedDDocJsonObj =
+        {[
+            {<<"_id">>, <<"_design/searchddoc">>},
+            {<<"_rev">>, ?b2l(couch_doc:rev_to_str(DDoc#doc.revs))},
+            {<<"indexes">>, {[
+                {<<"default">>, {[
+                    {<<"index">>, <<"function(doc){if(doc.blah){index(\\\"default\\\", doc.blah);}}">>}
+                ]}}
+            ]}}
+        ]},
+    UpdatedDDocJsonObj.
+
+ensureIndexIsOpened(DbName, DDocId, IndexName) ->
+    IndexWithName = getIndex(DbName, ?l2b(DDocId), ?l2b(IndexName)),
+    ?assertMatch(#index{sig=_Sig}, IndexWithName),
+    [#index{sig=Sig}] = [IndexWithName],
+    IndexPidsFromEts = get_index_by_sig_from_ets(?l2b(DDocId), Sig),
+    ?assert(length(IndexPidsFromEts) > 0),
+    Sig.
+
+ensureIndexIsClosed(DbName, DDocId, IndexName, Sig) ->
+    NotAnIndex = getIndex(DbName, ?l2b(DDocId), ?l2b(IndexName)),
+    ?assertEqual(ok, NotAnIndex),
+    (catch wait_for_indices_to_stop(?l2b(DbName))),
+    IndexPidsFromEts = get_index_by_sig_from_ets(?l2b(DDocId), Sig),
+    ?assertEqual(0, length(IndexPidsFromEts)),
+    ok.
+
+getIndex(DbName, DDocId, IndexName) ->
+    case fabric:open_doc(DbName, <<DDocId/binary>>, [ejson_body, ?ADMIN_CTX]) of
+        {ok, DDoc} ->
+            case dreyfus_index:design_doc_to_index(DDoc, IndexName) of
+                {ok, Index} ->
+                    Index;
+                _ ->
+                    ok
+            end;
+        _ ->
+            ok
+    end.
+
+get_index_by_sig_from_ets(DDocId, Sig) ->
+    Indexes = ets:match_object(
+        dreyfus_by_db, {'$1', {DDocId, Sig}}),
+    lists:foldl(fun({DbName, {_DDocId, _Sig}}, Acc) ->
+        case ets:lookup(dreyfus_by_sig, {DbName, Sig}) of
+            [{_, Pid}] -> [Pid| Acc];
```

Comment:
There is no guarantee that `dreyfus_by_sig` ets table contains Pid of an indexer process. There is a case when it would contain a Pid of an opener process [see here](https://github.com/cloudant-labs/dreyfus/blob/0739e2329877ea9222c8b00340c9d3b37b423220/src/dreyfus_index_manager.erl#L60).
In order to resolve this ambiguity you could use the approach [similar to the this](https://github.com/cloudant/couchdb/blob/6033c33bfaf8bb22c762eb01aedde57b6973fbd2/src/couch_index/test/couch_index_ddoc_updated_tests.erl#L127).